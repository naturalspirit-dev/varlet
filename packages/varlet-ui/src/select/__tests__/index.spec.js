const Select = require('../../../cjs/select').default
const { render } = require('@testing-library/vue')

test('test select', async () => {
	const wrapper = render(Select)
	console.log(wrapper)
})
