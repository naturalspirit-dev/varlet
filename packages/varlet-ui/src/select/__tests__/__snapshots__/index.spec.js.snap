// Vitest Snapshot v1

exports[`test select focus & blur methods 1`] = `
"<div class=\\"var-select var--box\\">
  <div class=\\"var-select__controller var-select--focus\\">
    <div class=\\"var-select__icon\\"></div>
    <div class=\\"var-select__wrap\\">
      <div class=\\"var-menu var-select__menu\\" var-select-cover=\\"\\">
        <div class=\\"var-select__select\\" style=\\"text-align: left;\\"><span></span><i class=\\"var-icon var-icon--set var-icon-menu-down var-select__arrow var-select--arrow-rotate\\" style=\\"transition: transform 300ms;\\" var-select-cover=\\"\\"></i></div><label class=\\"var-select__placeholder var--ellipsis var-select--focus var-select--placeholder-hint\\"></label>
        <!--teleport start-->
        <!--teleport end-->
      </div>
    </div>
    <div class=\\"var-select__icon\\">
      <!--v-if-->
    </div>
  </div>
  <div class=\\"var-select__line\\">
    <div class=\\"var-select__dot var-select--spread\\"></div>
  </div>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
</div>"
`;

exports[`test select focus & blur methods 2`] = `
"<div class=\\"var-select var--box\\">
  <div class=\\"var-select__controller\\">
    <div class=\\"var-select__icon\\"></div>
    <div class=\\"var-select__wrap\\">
      <div class=\\"var-menu var-select__menu\\" var-select-cover=\\"\\">
        <div class=\\"var-select__select\\" style=\\"text-align: left;\\"><span></span><i class=\\"var-icon var-icon--set var-icon-menu-down var-select__arrow\\" style=\\"transition: transform 300ms;\\" var-select-cover=\\"\\"></i></div><label class=\\"var-select__placeholder var--ellipsis\\"></label>
        <!--teleport start-->
        <!--teleport end-->
      </div>
    </div>
    <div class=\\"var-select__icon\\">
      <!--v-if-->
    </div>
  </div>
  <div class=\\"var-select__line\\">
    <div class=\\"var-select__dot\\"></div>
  </div>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
</div>"
`;

exports[`test select hint to be false 1`] = `
"<div class=\\"var-select var--box\\">
  <div class=\\"var-select__controller\\">
    <div class=\\"var-select__icon var-select--non-hint\\"></div>
    <div class=\\"var-select__wrap var-select--non-hint\\">
      <div class=\\"var-menu var-select__menu\\" var-select-cover=\\"\\">
        <div class=\\"var-select__select\\" style=\\"text-align: left;\\"><span></span><i class=\\"var-icon var-icon--set var-icon-menu-down var-select__arrow\\" style=\\"transition: transform 300ms;\\" var-select-cover=\\"\\"></i></div><label class=\\"var-select__placeholder var--ellipsis var-select--placeholder-hidden var-select--placeholder-non-hint\\"></label>
        <!--teleport start-->
        <!--teleport end-->
      </div>
    </div>
    <div class=\\"var-select__icon var-select--non-hint\\">
      <!--v-if-->
    </div>
  </div>
  <div class=\\"var-select__line\\">
    <div class=\\"var-select__dot\\"></div>
  </div>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
</div>"
`;

exports[`test select validation 1`] = `
"<div class=\\"var-select var--box\\">
  <div class=\\"var-select__controller var-select--error\\">
    <div class=\\"var-select__icon\\"></div>
    <div class=\\"var-select__wrap\\">
      <div class=\\"var-menu var-select__menu\\" var-select-cover=\\"\\">
        <div class=\\"var-select__select var-select--error\\" style=\\"text-align: left;\\"><span></span><i class=\\"var-icon var-icon--set var-icon-menu-down var-select__arrow\\" style=\\"transition: transform 300ms;\\" var-select-cover=\\"\\"></i></div><label class=\\"var-select__placeholder var--ellipsis var-select--error\\"></label>
        <!--teleport start-->
        <!--teleport end-->
      </div>
    </div>
    <div class=\\"var-select__icon\\">
      <!--v-if-->
    </div>
  </div>
  <div class=\\"var-select__line var-select--line-error\\">
    <div class=\\"var-select__dot var-select--line-error\\"></div>
  </div>
  <transition-stub>
    <div class=\\"var-form-details\\">
      <div class=\\"var-form-details__message\\">您必须选择一个</div>
      <div class=\\"var-form-details__length\\"></div>
    </div>
  </transition-stub>
</div>"
`;
